app/src/test/java/com/x8bit/bitwarden/data/autofill/accessibility/manager/LauncherPackageNameManagerTest.kt

```diff
@@ -0,0 +1,59 @@
+package com.x8bit.bitwarden.data.autofill.accessibility.manager
+
+import android.content.pm.ActivityInfo
+import android.content.pm.PackageManager
+import android.content.pm.ResolveInfo
+import io.mockk.clearMocks
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import java.time.Clock
+import java.time.Instant
+import java.time.ZoneOffset
+
+class LauncherPackageNameManagerTest {
+    private val packageManager: PackageManager = mockk()
+
+    private val launcherPackageNameManager: LauncherPackageNameManager =
+        LauncherPackageNameManagerImpl(
+            clock = FIXED_CLOCK,
+            packageManager = packageManager,
+        )
+
+    @Suppress("MaxLineLength")
+    @Test
+    fun `launcherPackages should populate cache on first attempt and use cached value for second attempt`() {
```

Comment:
Can we add either more to this test or a completely separate test that shows how the clock factors into all this?